filecoin_test.go

```diff
@@ -63,7 +64,7 @@ func TestFilecoinBasic(t *testing.T) {
 
 	// TODO: can we do better than these type asserts?
 	node := builder.Build().(*hamt.Node)
-	nodeRepr := node.Substrate().(hamt.HashMapNode).Representation()
+	nodeRepr := node.Substrate().(schema.TypedNode).Representation()
```

Comment:
maybe `Substrate()` should be returning a `schema.TypedNode` anwyay?

Reply:
or maybe not .. I guess taking the "lens" approach, the substrate could be anything.

But what you could do below is `enc, err := ipld.Encode(node.Substrate(), dagcbor.Encode)` and it'll take care of the `Representation()` step as required.